src/redist/targets/GenerateLayout.targets

```diff
@@ -7,14 +7,19 @@
 
   <PropertyGroup>
     <!-- Blob storage directories are not stabilized, so these must refer to a package that does not stabilize -->
-    <AspNetCoreBlobVersion>$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
-    <CoreSetupBlobVersion>$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
+    <!-- In source build, the layout does match, so use the runtime package versions rather than the VS redist versions
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftAspNetCoreAppRuntimePackageVersion)</AspNetCoreBlobVersion>
+    <AspNetCoreBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonAspNetCoreSharedFrameworkx6470PackageVersion)</AspNetCoreBlobVersion>
+    <CoreSetupBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(MicrosoftNETCoreAppRuntimePackageVersion)</CoreSetupBlobVersion>
+    <CoreSetupBlobVersion Condition=" '$(DotNetBuildFromSource)' != 'true' ">$(VSRedistCommonNetCoreSharedFrameworkx6470PackageVersion)</CoreSetupBlobVersion>
     <WindowsDesktopBlobVersion>$(VSRedistCommonWindowsDesktopSharedFrameworkx6470PackageVersion)</WindowsDesktopBlobVersion>
 
-    <!-- Change these individually to or $(CoreSetupBlobVersion), $(AspNetCoreBlobVersion), or appropriate fixed version depending if corresponding .Ref packages are unpinned. -->
-    <NETCoreAppTargetingPackBlobVersion>$(CoreSetupBlobVersion)</NETCoreAppTargetingPackBlobVersion>
-    <AspNetCoreTargetingPackBlobVersion>$(MicrosoftAspNetCoreAppRefInternalPackageVersion)</AspNetCoreTargetingPackBlobVersion>
-    <WindowsDesktopTargetingPackBlobVersion>$(WindowsDesktopBlobVersion)</WindowsDesktopTargetingPackBlobVersion>
+    <!-- Because the ref packs are not serviced in every build, the blob location may vary -->
+    <NETCoreAppTargetingPackBlobVersion Condition=" '$(DotNetBuildFromSource)' == 'true' ">$(CoreSetupBlobVersion)</NETCoreAppTargetingPackBlobVersion>
```

Comment:
@dseefeld This bit I have the biggest question about. When building in source build, what is the name of the ref pack blob directory? Does it diverge from the runtime pack blob directory?

Reply:
I'm not sure I understand the question.  This is setting the Version, not the directory, right?

Reply:
It's setting the directory that the ref pack will appear in, which is related to the version. E.g.:

`dotnet/Runtime/<ref pack full version>/dotnet-targeting-pack-<ref pack stable version>/`

Reply:
Looking at a source-build build, it appears that the directories are named with the same version.